client/components/PrettyPrint/PrettyPrint.jsx

```diff
@@ -0,0 +1,76 @@
+import { h, Component } from "preact";
+import cx from "classnames";
+import Prism from "prismjs";
+
+// Import prism plugins
+import "prismjs/plugins/line-numbers/prism-line-numbers";
+import "prismjs/plugins/line-highlight/prism-line-highlight";
+
+import "./prettyprint.css";
+
+const BASE_CLASS = "pretty-print";
+
+export default class PrettyPrint extends Component {
+  constructor(...args) {
+    super(...args);
+    this.highlight = this.highlight.bind(this);
```

Comment:
I see the only place of invoking `highlight` is in `componentDidMount` and it is invoked as `this.highlight()`, and `componentDidMount` is already auto-bound to `this`. So invoking it as 

```js
const x = this.highlight;
x();
```

has no this bound to `highlight`. But invoking it as -

```js
this.highlight();
```

has this binding in highlight. So it should work without this binding.